File: webui/src/types/session.ts
/**
 * Session types — mirror Pydantic shapes in `controllers/sessions.py`.
 *
 * snake_case is intentional: matches wire format from the backend.
 */

/** A single session row, as returned by GET / list endpoints. */
export interface SessionInfo {
  /** Server-assigned session UUID. */
  session_id: string;
  /** Origin: `'workflow'` for DAG executions, `'conversation'` for chats. */
  source: string;
  /** Owning workflow execution UUID; null for conversation-originated. */
  execution_id: string | null;
  /** Owning workflow node ID; null for conversation-originated. */
  node_id: string | null;
  /** Agent name bound to the session; null when unbound. */
  agent: string | null;
  /** Lifecycle state: `'active' | 'inactive' | 'cleaned_up' | 'unknown'`. */
  status: string;
  /** ISO-8601 UTC timestamp of session creation. */
  created_at: string | null;
}

/** Envelope returned by GET /api/v1/sessions. */
export interface SessionListResponse {
  sessions: SessionInfo[];
  total: number;
  active_count: number;
}

/** Request body for POST /api/v1/sessions/cleanup. */
export interface CleanupRequest {
  /** Cleanup sessions attached to this execution UUID. */
  execution_id?: string;
  /** When true, clean up all expired sessions server-wide. */
  all_expired?: boolean;
}

/** Response body for POST /api/v1/sessions/cleanup. */
export interface CleanupResponse {
  /** Number of sessions successfully removed. */
  cleaned: number;
  /** Number of sessions that failed to remove. */
  failed: number;
  /** Per-session status map (`session_id → "ok" | "<error message>"`). */
  details: Record<string, string>;
}

File: src/main/framework/schemas/session.py
"""Session API schemas — request/response models for the sessions controller.

These Pydantic V2 models were originally declared inline in
``controllers/sessions.py`` and are now extracted here as part of the
PHASE 5 directory reorganization. Field names, types, defaults, and
validators are preserved verbatim — only the import path changes.

Endpoints served by these schemas (see ``controllers/sessions.py``):

* ``GET /api/v1/sessions`` → ``SessionListResponse``
* ``GET /api/v1/sessions/{session_id}`` → ``SessionInfo``
* ``DELETE /api/v1/sessions/{session_id}`` → dict (no response model)
* ``POST /api/v1/sessions/cleanup`` → body ``CleanupRequest``,
  response ``CleanupResponse``
"""

from __future__ import annotations

from pydantic import BaseModel


class SessionInfo(BaseModel):
    session_id: str
    source: str # "workflow" | "conversation"
    execution_id: str | None = None
    node_id: str | None = None
    agent: str | None = None
    status: str # "active" | "inactive" | "cleaned_up" | "unknown"
    created_at: str | None = None


class SessionListResponse(BaseModel):
    sessions: list[SessionInfo]
    total: int
    active_count: int


class CleanupRequest(BaseModel):
    execution_id: str | None = None
    all_expired: bool = False


class CleanupResponse(BaseModel):
    cleaned: int
    failed: int
    details: dict[str, str]
